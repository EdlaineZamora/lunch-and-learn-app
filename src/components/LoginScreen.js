import React, { Component } from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  View,
  Image,
  Dimensions,
  TextInput,
  Button,
  TouchableOpacity
} from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import User from './user/User.js'
import Password from './password/Password.js'

const { width, height } = Dimensions.get("window");

const mark = require("./logo.jpg");

export default class LoginScreen extends Component {
  render() {
    return (
      <View style={styles.container}>
          <View style={styles.markWrap}>
            <Image source={mark} style={styles.mark} resizeMode="contain" />
          </View>
          <View style={styles.wrapper}>
            <User />
            <Password />
            <TouchableOpacity activeOpacity={.5}>
              <View style={styles.button}>
                <Text style={styles.buttonText}>Logar</Text>
              </View>
            </TouchableOpacity>
          </View>
          <View style={styles.container}>
            <View style={styles.signupWrap}>
              <Text style={styles.accountText}>NÃ£o tem uma conta?</Text>
              <TouchableOpacity activeOpacity={.5}>
                <View>
                  <Text style={styles.signupLinkText}>Cadastrar</Text>
                </View>
              </TouchableOpacity>
            </View>
          </View>
      </View>
    );
  }
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  markWrap: {
    flex: 1,
    paddingVertical: 30,
  },
  mark: {
    width: null,
    height: null,
    flex: 1,
  },
  background: {
    width,
    height,
  },
  wrapper: {
    paddingVertical: 30,
  },
  inputPasswordWrap: {
    flexDirection: "row",
    marginVertical: 10,
    height: 40,
    borderBottomWidth: 1,
    borderBottomColor: "#CCC"
  },
  iconWrap: {
    paddingHorizontal: 7,
    alignItems: "center",
    justifyContent: "center",
  },
  icon: {
    height: 20,
    width: 20,
  },
  input: {
    flex: 1,
    paddingHorizontal: 10,
  },
  button: {
    backgroundColor: "#3c6fc1",
    paddingVertical: 20,
    alignItems: "center",
    justifyContent: "center",
    marginTop: 30,
  },
  buttonText: {
    color: "#FFF",
    fontSize: 18,
  },
  signupWrap: {
    backgroundColor: "transparent",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    color: 'black'
  },
  accountText: {
    color: "#000"
  },
  signupLinkText: {
    color: "#000",
    marginLeft: 5,
  }
});