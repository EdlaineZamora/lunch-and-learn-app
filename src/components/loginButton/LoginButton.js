import React, { Component } from 'react';
import { View, TouchableOpacity, Text } from 'react-native';
import Styles from './Styles.js';

class LoginButton extends Component {
    render() {
        return (
            <View>
            <TouchableOpacity activeOpacity={.5}>
                <View style={Styles.button}>
                    <Text style={Styles.buttonText}>Logar</Text>
                </View>
            </TouchableOpacity>
            </View>
        );
    }
}

export default LoginButton;