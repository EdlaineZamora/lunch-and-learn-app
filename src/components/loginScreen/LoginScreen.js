import React, { Component } from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  View,
  Image,
  Dimensions,
  TextInput,
  Button,
  TouchableOpacity
} from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import User from '../user/User.js';
import Password from '../password/Password.js';
import NewAccount from '../newAccount/NewAccount.js';
import LoginButton from '../loginButton/LoginButton.js';
import Styles from './Styles.js';

const mark = require("../logo.jpg");

export default class LoginScreen extends Component {
  render() {
    return (
      <View style={Styles.container}>
          <View style={Styles.markWrap}>
            <Image source={mark} style={Styles.mark} resizeMode="contain" />
          </View>
          <View style={Styles.wrapper}>
            <User />
            <Password />
            <LoginButton/>
          </View>
          <NewAccount />
      </View>
    );
  }
}