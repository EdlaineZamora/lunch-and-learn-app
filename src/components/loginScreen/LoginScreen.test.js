import React from 'react';
import { shallow } from 'enzyme';
import LoginScreen from './LoginScreen';
import LoginButton from '../loginButton/LoginButton';
import { TouchableOpacity } from 'react-native';

describe('<LoginScreen>', () => {
  let loginWrapper;

  beforeAll(() => {
    loginWrapper = shallow(<LoginScreen />);
  });

  it('renders correctly', () => {
    expect(loginWrapper).toMatchSnapshot();
  });

  describe('when LoginButton is pressed', () => {
    it('should show an error message if credentials are empty', () => {
      const button = loginWrapper.find(LoginButton).dive();

      button.find(TouchableOpacity).simulate('press');
      loginWrapper.update();
      
      expect(loginWrapper.find('#error-message').length).toBe(1);
    });
  })
});