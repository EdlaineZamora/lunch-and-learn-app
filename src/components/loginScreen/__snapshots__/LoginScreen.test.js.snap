// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginScreen> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "paddingVertical": 30,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/components/loginScreen/logo.jpg",
        }
      }
      style={
        Object {
          "flex": 1,
          "height": null,
          "width": null,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Lunch & Learn
    </Text>
  </View>
  <View
    style={
      Object {
        "paddingVertical": 30,
      }
    }
  >
    <LoginTextInput
      changeText={[Function]}
      iconName="ios-person"
      isSecure={false}
      placeholder="UsuÃ¡rio"
      value=""
    />
    <LoginTextInput
      changeText={[Function]}
      iconName="ios-lock"
      isSecure={true}
      placeholder="Senha"
      value=""
    />
    <LoginButton
      onPressButton={[Function]}
    />
  </View>
  <NewAccount />
</View>
`;
